kedro-org/kedro-viz
# [KED-3004] Search MVP experiment tracking

## Description

resolves [KED-3004](https://jira.quantumblack.com/browse/KED-3004).

This PR is the work in setting up the search for experiment tracking runslist. This will now allow users to search through the runs list and return results with runs that either the 'Title', 'gitsha' and 'notes' field matches the search value.

![image](https://user-images.githubusercontent.com/27775658/150325108-3ecda9fb-ec60-44ec-9456-3643d9d298da.png)

![image](https://user-images.githubusercontent.com/27775658/150325255-e06ef563-f0d7-4cd9-bf4b-826ddfefdf87.png)


## Development notes

1. I have refactored the existing search bar from a tailored component to the node-list to a general search bar, and set up related tests.
2. I have refactored existing runs-list and runs-list card setup with the search Logic, and added in tests related to the search feature.
3. I have set up a new test suite for runs-list component as currently there isn't any test suites for that component. 

## QA notes

Please check out the branch and QA the search. I will also sync up with design (@gabrielcomym) to go through the latest implementation. 

## Checklist
